catkin_ws/src/state_estimation/src/state_aggregator.py

```diff
@@ -29,8 +33,20 @@ def update_state(_):
             break
     for sensor in sensor_priorities["z"]:
         if sensor.isActive():
-            z = sensor.z
-            break
+            if sensor.sensor_name == "depth_sensor":
+                ds_z = sensor.z
+                if dvl_z is not None:
+                    offset = math.abs(ds_z - dvl_z)
+                    i += 1
+                    ds_z -= math.exp(-i)*offset
+                z = ds_z
+            else:
+                dvl_z = sensor.z
+                if ds_z is not None:
+                    offset = math.abs(ds_z - dvl_z)
```

Comment:
I think this wouldnt work again. Here you're taking the absolute difference between the last depth sensor reading and the current dvl reading, and applying that offset to the DVL. The problem with that is that 1. the DVL reading will always be equal to the depth sensor, since the offset that is calculated is not saved and re-used, so essentially the DVL is always being offset by an amount so it is equal to the last depth sensor value. What we want is that the first time the DVL is available and the depth sensor is not, we save the offset, and then on later iterations we always apply that offset that was calculated the first time. Also, using an absolute value doesnt work, when you subtract an absolute value from the DVL's reading you're making the assumption that the DVL's reading will always be greater than the depth sensor's last reading, which isn't true.